qt/python/mantidqtinterfaces/mantidqtinterfaces/Muon/GUI/Common/phase_table_widget/phase_table_model.py

```diff
@@ -0,0 +1,132 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2024 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+from mantidqtinterfaces.Muon.GUI.Common.ADSHandler.workspace_naming import (
+    get_run_numbers_as_string_from_workspace_name,
+    get_base_data_directory,
+    get_phase_table_workspace_name,
+)
+from mantidqtinterfaces.Muon.GUI.Common.ADSHandler.muon_workspace_wrapper import MuonWorkspaceWrapper
+from mantidqtinterfaces.Muon.GUI.Common.utilities.algorithm_utils import run_CalMuonDetectorPhases
+from mantid import AlgorithmManager
+
+
+class PhaseTableModel:
+    def __init__(self, context):
+        self._context = context
+        self._current_alg = None
+        self._phasequad = None
+        self._new_table_name = ""
+        self._calculation_thread = None
+        self._phasequad_calculation_thread = None
+
+    @property
+    def context(self):
+        return self._context
+
+    @property
+    def phase_context(self):
+        return self._context.phase_context
+
+    @property
+    def group_pair_context(self):
+        return self._context.group_pair_context
+
+    @property
+    def current_alg(self):
+        return self._current_alg
+
+    @current_alg.setter
+    def current_alg(self, value):
+        self._current_alg = value
+
+    @property
+    def phasequad(self):
+        return self._phasequad
+
+    @phasequad.setter
+    def phasequad(self, value):
+        self._phasequad = value
+
+    @property
+    def new_table_name(self):
+        return self._new_table_name
+
+    @new_table_name.setter
+    def new_table_name(self, value):
+        self._new_table_name = value
+
+    @property
+    def calculation_thread(self):
+        return self._calculation_thread
+
+    @calculation_thread.setter
+    def calculation_thread(self, value):
+        self._calculation_thread = value
+
+    @property
+    def phasequad_calculation_thread(self):
+        return self._calculation_thread
```

Comment:
Should these `phasequad_calculation_thread` properties be using the `self._phasequad_calculation_thread` variable?